Some updates:

- The `test_saved_model` tests are all fixed now.
- The `ImageClassifier` tests used `EfficientNetV2` so I also ported it to `keras_core` in this PR. This PR has gotten pretty big on its own but I have tried to keep commits organized; each commit ports one component (e.g. [fe96bbf](https://github.com/ianstenbit/keras-cv/pull/2/commits/fe96bbf8e62be7a914fd8672ca80e3ba66c5f5a5) ports `EfficientNet` and its tests). An alternative is to put each commit in it's own PR. Let me know if you prefer to divide it into multiple PRs and reviewing each component separately.
- There's still one failing test in the `ImageClassifier` test suite:

<details> <summary>Error Log</summary>

```
========================================================================= test session starts ==========================================================================
platform linux -- Python 3.10.9, pytest-7.3.1, pluggy-1.0.0
rootdir: /usr/local/google/home/tirthp/oss/keras-cv
configfile: setup.cfg
plugins: cov-4.1.0, anyio-3.6.2, xdist-3.3.1
collected 81 items                                                                                                                                                     

keras_cv/models/backbones/resnet_v1/resnet_v1_backbone_presets_test.py .........s                                                                                [ 12%]
keras_cv/models/backbones/resnet_v1/resnet_v1_backbone_test.py ..............                                                                                    [ 29%]
keras_cv/models/backbones/resnet_v2/resnet_v2_backbone_presets_test.py .........s                                                                                [ 41%]
keras_cv/models/backbones/resnet_v2/resnet_v2_backbone_test.py ..........                                                                                        [ 54%]
keras_cv/models/backbones/efficientnet_v2/efficientnet_v2_backbone_presets_test.py ............s                                                                 [ 70%]
keras_cv/models/backbones/efficientnet_v2/efficientnet_v2_backbone_test.py ..........                                                                            [ 82%]
keras_cv/models/classification/image_classifier_test.py ....F.........                                                                                           [100%]

=============================================================================== FAILURES ===============================================================================
_________________________________________________________________ ImageClassifierTest.test_saved_model _________________________________________________________________

self = <keras_cv.models.classification.image_classifier_test.ImageClassifierTest testMethod=test_saved_model>

    @pytest.mark.large  # Saving is slow, so mark these large.
    def test_saved_model(self):
        model = ImageClassifier(
            backbone=ResNet18V2Backbone(),
            num_classes=2,
        )
        model_output = model(self.input_batch)
        save_path = os.path.join(self.get_temp_dir(), "image_classifier.keras")
        model.save(save_path)
>       restored_model = keras.models.load_model(save_path)

keras_cv/models/classification/image_classifier_test.py:90: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../virtualenvs/keras-cv-dev/lib/python3.10/site-packages/keras/saving/saving_api.py:204: in load_model
    return saving_lib.load_model(
../virtualenvs/keras-cv-dev/lib/python3.10/site-packages/keras/saving/saving_lib.py:277: in load_model
    raise e
../virtualenvs/keras-cv-dev/lib/python3.10/site-packages/keras/saving/saving_lib.py:242: in load_model
    model = deserialize_keras_object(
../virtualenvs/keras-cv-dev/lib/python3.10/site-packages/keras/saving/serialization_lib.py:502: in deserialize_keras_object
    instance = cls.from_config(inner_config)
keras_cv/models/task.py:54: in from_config
    return cls(**config)
keras_cv/models/classification/image_classifier.py:99: in __init__
    super().__init__(
keras_cv/models/task.py:28: in __init__
    super().__init__(*args, **kwargs)
../virtualenvs/keras-cv-dev/lib/python3.10/site-packages/keras_core/utils/tracking.py:21: in wrapper
    return fn(*args, **kwargs)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <[AttributeError("'ImageClassifier' object has no attribute 'name'") raised in repr()] ImageClassifier object at 0x7f9912f7e7a0>
inputs = <KerasTensor shape=(None, None, None, 3), dtype=float32, name=keras_tensor_10128>
outputs = <KerasTensor shape=(None, 2), dtype=float32, name=keras_tensor_10198>, name = 'image_classifier_4', kwargs = {'trainable': True}

    @tracking.no_automatic_dependency_tracking
    def __init__(self, inputs, outputs, name=None, **kwargs):
        if isinstance(inputs, dict):
            for k, v in inputs.items():
                if not isinstance(v, backend.KerasTensor):
                    raise ValueError(
                        "When providing `inputs` as a dict, all values in the "
                        f"dict must be KerasTensors. Received: inputs={inputs} "
                        f"including invalid value {v} of type {type(v)}"
                    )
                if k != v.name:
                    # TODO: maybe make this a warning
                    raise ValueError(
                        "When providing `inputs` as a dict, all keys in the "
                        "dict must match the names of the corresponding "
                        f"tensors. Received key '{k}' mapping to value {v} "
                        f"which has name '{v.name}'. Change the tensor name to "
                        f"'{k}' (via `Input(..., name='{k}')`)"
                    )
        elif isinstance(inputs, (list, tuple)):
            for x in inputs:
                if not isinstance(x, backend.KerasTensor):
                    raise ValueError(
                        "When providing `inputs` as a list/tuple, all values "
                        f"in the list/tuple must be KerasTensors. Received: "
                        f"inputs={inputs} including invalid value {x} of type "
                        f"{type(x)}"
                    )
        elif not isinstance(inputs, backend.KerasTensor):
            raise ValueError(
                f"Unrecognized type for `inputs`: {inputs} "
                f"(of type {type(inputs)})"
            )
        if isinstance(outputs, dict):
            for k, v in outputs.items():
                if not isinstance(v, backend.KerasTensor):
                    raise ValueError(
                        "When providing `outputs` as a dict, all values in the "
                        f"dict must be KerasTensors. Received: "
                        f"outputs={outputs} including invalid value {v} of "
                        f"type {type(v)}"
                    )
        elif isinstance(outputs, (list, tuple)):
            for x in outputs:
                if not isinstance(x, backend.KerasTensor):
                    raise ValueError(
                        "When providing `outputs` as a list/tuple, all values "
                        f"in the list/tuple must be KerasTensors. Received: "
                        f"outputs={outputs} including invalid value {x} of "
                        f"type {type(x)}"
                    )
        elif not isinstance(outputs, backend.KerasTensor):
            raise ValueError(
                f"Unrecognized type for `outputs`: {outputs} "
                f"(of type {type(outputs)})"
            )
    
>       Function.__init__(self, inputs, outputs, name=name, **kwargs)
E       TypeError: Function.__init__() got an unexpected keyword argument 'trainable'

../virtualenvs/keras-cv-dev/lib/python3.10/site-packages/keras_core/models/functional.py:146: TypeError
========================================================================== slowest durations ===========================================================================
47.83s call     keras_cv/models/backbones/efficientnet_v2/efficientnet_v2_backbone_test.py::EfficientNetV2BackboneTest::test_saved_alias_model
42.94s call     keras_cv/models/backbones/efficientnet_v2/efficientnet_v2_backbone_test.py::EfficientNetV2BackboneTest::test_saved_model
42.42s call     keras_cv/models/backbones/resnet_v2/resnet_v2_backbone_presets_test.py::ResNetV2PresetFullTest::test_load_resnetv2
30.48s call     keras_cv/models/backbones/efficientnet_v2/efficientnet_v2_backbone_presets_test.py::EfficientNetV2PresetFullTest::test_load_efficientnet_efficientnetv2_l
27.71s call     keras_cv/models/backbones/efficientnet_v2/efficientnet_v2_backbone_presets_test.py::EfficientNetV2PresetFullTest::test_load_efficientnet_efficientnetv2_m
22.07s call     keras_cv/models/backbones/resnet_v2/resnet_v2_backbone_test.py::ResNetV2BackboneTest::test_saved_alias_model
18.61s call     keras_cv/models/backbones/efficientnet_v2/efficientnet_v2_backbone_presets_test.py::EfficientNetV2PresetFullTest::test_load_efficientnet_efficientnetv2_b0
18.06s call     keras_cv/models/backbones/efficientnet_v2/efficientnet_v2_backbone_presets_test.py::EfficientNetV2PresetFullTest::test_load_efficientnet_efficientnetv2_b2
17.41s call     keras_cv/models/backbones/efficientnet_v2/efficientnet_v2_backbone_presets_test.py::EfficientNetV2PresetFullTest::test_load_efficientnet_efficientnetv2_b3
16.65s call     keras_cv/models/classification/image_classifier_test.py::ImageClassifierTest::test_classifier_fit_jit_compile_false
15.92s call     keras_cv/models/backbones/efficientnet_v2/efficientnet_v2_backbone_presets_test.py::EfficientNetV2PresetFullTest::test_efficientnet_feature_extractor
15.21s call     keras_cv/models/backbones/efficientnet_v2/efficientnet_v2_backbone_test.py::EfficientNetV2BackboneTest::test_valid_call_alias_model_with_rescaling
14.93s call     keras_cv/models/backbones/efficientnet_v2/efficientnet_v2_backbone_test.py::EfficientNetV2BackboneTest::test_valid_call_with_rescaling
12.70s call     keras_cv/models/backbones/efficientnet_v2/efficientnet_v2_backbone_test.py::EfficientNetV2BackboneTest::test_valid_call
10.14s call     keras_cv/models/backbones/efficientnet_v2/efficientnet_v2_backbone_presets_test.py::EfficientNetV2PresetFullTest::test_load_efficientnet_efficientnetv2_s
9.98s call     keras_cv/models/classification/image_classifier_test.py::ImageClassifierTest::test_classifier_fit_jit_compile_true
9.65s call     keras_cv/models/backbones/resnet_v1/resnet_v1_backbone_test.py::ResNetBackboneTest::test_saved_alias_model
9.32s call     keras_cv/models/backbones/resnet_v1/resnet_v1_backbone_presets_test.py::ResNetPresetFullTest::test_load_resnet
7.85s call     keras_cv/models/backbones/efficientnet_v2/efficientnet_v2_backbone_presets_test.py::EfficientNetV2PresetFullTest::test_load_efficientnet_efficientnetv2_s_imagenet
7.37s call     keras_cv/models/backbones/resnet_v2/resnet_v2_backbone_presets_test.py::ResNetV2PresetSmokeTest::test_applications_model_output
7.29s call     keras_cv/models/backbones/efficientnet_v2/efficientnet_v2_backbone_test.py::EfficientNetV2BackboneTest::test_create_backbone_model_from_alias_model
7.19s call     keras_cv/models/backbones/resnet_v1/resnet_v1_backbone_presets_test.py::ResNetPresetSmokeTest::test_applications_model_output
5.92s call     keras_cv/models/backbones/efficientnet_v2/efficientnet_v2_backbone_test.py::EfficientNetV2BackboneTest::test_create_backbone_model_with_level_config
5.34s call     keras_cv/models/backbones/efficientnet_v2/efficientnet_v2_backbone_test.py::EfficientNetV2BackboneTest::test_application_variable_input_channels_four_channels
5.17s call     keras_cv/models/backbones/resnet_v2/resnet_v2_backbone_test.py::ResNetV2BackboneTest::test_saved_model
4.98s call     keras_cv/models/classification/image_classifier_test.py::ImageClassifierPresetSmokeTest::test_efficientnet_v2_preset_efficientnetv2_b0_imagenet_classifier
4.93s call     keras_cv/models/backbones/efficientnet_v2/efficientnet_v2_backbone_test.py::EfficientNetV2BackboneTest::test_application_variable_input_channels_one_channel
4.82s call     keras_cv/models/backbones/efficientnet_v2/efficientnet_v2_backbone_presets_test.py::EfficientNetV2PresetFullTest::test_load_efficientnet_efficientnetv2_b1_imagenet
4.75s call     keras_cv/models/backbones/efficientnet_v2/efficientnet_v2_backbone_presets_test.py::EfficientNetV2PresetFullTest::test_load_efficientnet_efficientnetv2_b2_imagenet
4.44s call     keras_cv/models/backbones/efficientnet_v2/efficientnet_v2_backbone_presets_test.py::EfficientNetV2PresetFullTest::test_load_efficientnet_efficientnetv2_b1
3.92s call     keras_cv/models/backbones/resnet_v1/resnet_v1_backbone_test.py::ResNetBackboneTest::test_saved_model
3.90s call     keras_cv/models/backbones/resnet_v1/resnet_v1_backbone_test.py::ResNetBackboneTest::test_specific_arch_forward_pass_101
3.84s call     keras_cv/models/backbones/resnet_v1/resnet_v1_backbone_presets_test.py::ResNetPresetSmokeTest::test_backbone_output_with_weights
3.76s call     keras_cv/models/backbones/resnet_v2/resnet_v2_backbone_presets_test.py::ResNetV2PresetSmokeTest::test_backbone_output_preset_with_weights
3.74s call     keras_cv/models/backbones/resnet_v2/resnet_v2_backbone_presets_test.py::ResNetV2PresetSmokeTest::test_applications_model_output_with_preset
3.71s call     keras_cv/models/backbones/efficientnet_v2/efficientnet_v2_backbone_presets_test.py::EfficientNetV2PresetFullTest::test_load_efficientnet_efficientnetv2_b0_imagenet
3.12s call     keras_cv/models/classification/image_classifier_test.py::ImageClassifierPresetSmokeTest::test_backbone_preset_call_preset_no_weights
3.10s call     keras_cv/models/backbones/resnet_v2/resnet_v2_backbone_test.py::ResNetV2BackboneTest::test_valid_call_applications_model
3.07s call     keras_cv/models/backbones/resnet_v2/resnet_v2_backbone_presets_test.py::ResNetV2PresetSmokeTest::test_backbone_output_preset_no_weights
2.85s call     keras_cv/models/backbones/resnet_v1/resnet_v1_backbone_test.py::ResNetBackboneTest::test_specific_arch_forward_pass_152
2.49s call     keras_cv/models/classification/image_classifier_test.py::ImageClassifierPresetSmokeTest::test_classifier_preset_call
1.99s call     keras_cv/models/classification/image_classifier_test.py::ImageClassifierPresetSmokeTest::test_backbone_preset_call_preset_with_weights
1.97s call     keras_cv/models/classification/image_classifier_test.py::ImageClassifierPresetSmokeTest::test_backbone_preset_weight_loading
1.87s call     keras_cv/models/backbones/resnet_v2/resnet_v2_backbone_test.py::ResNetV2BackboneTest::test_application_variable_input_channels_four_channels
1.78s call     keras_cv/models/backbones/resnet_v2/resnet_v2_backbone_test.py::ResNetV2BackboneTest::test_application_variable_input_channels_one_channel
1.76s call     keras_cv/models/backbones/resnet_v2/resnet_v2_backbone_test.py::ResNetV2BackboneTest::test_create_backbone_model_from_alias_model
1.59s call     keras_cv/models/backbones/resnet_v1/resnet_v1_backbone_presets_test.py::ResNetPresetSmokeTest::test_applications_model_output_with_preset
1.44s call     keras_cv/models/backbones/resnet_v2/resnet_v2_backbone_test.py::ResNetV2BackboneTest::test_valid_call_with_rescaling
1.37s call     keras_cv/models/backbones/resnet_v2/resnet_v2_backbone_test.py::ResNetV2BackboneTest::test_valid_call
1.02s call     keras_cv/models/backbones/resnet_v1/resnet_v1_backbone_presets_test.py::ResNetPresetSmokeTest::test_backbone_output
1.01s call     keras_cv/models/backbones/resnet_v1/resnet_v1_backbone_test.py::ResNetBackboneTest::test_specific_arch_forward_pass_50
0.98s call     keras_cv/models/classification/image_classifier_test.py::ImageClassifierTest::test_saved_model
0.97s call     keras_cv/models/backbones/resnet_v1/resnet_v1_backbone_test.py::ResNetBackboneTest::test_valid_call_applications_model
0.94s call     keras_cv/models/backbones/resnet_v2/resnet_v2_backbone_test.py::ResNetV2BackboneTest::test_create_backbone_model_with_level_config
0.84s call     keras_cv/models/backbones/resnet_v1/resnet_v1_backbone_test.py::ResNetBackboneTest::test_application_variable_input_channels_one_channel
0.82s call     keras_cv/models/backbones/resnet_v1/resnet_v1_backbone_test.py::ResNetBackboneTest::test_create_backbone_model_from_alias_model
0.75s call     keras_cv/models/backbones/resnet_v1/resnet_v1_backbone_test.py::ResNetBackboneTest::test_application_variable_input_channels_four_channels
0.65s call     keras_cv/models/backbones/resnet_v1/resnet_v1_backbone_test.py::ResNetBackboneTest::test_specific_arch_forward_pass_18
0.61s call     keras_cv/models/backbones/resnet_v1/resnet_v1_backbone_test.py::ResNetBackboneTest::test_valid_call_with_rescaling
0.55s call     keras_cv/models/backbones/resnet_v1/resnet_v1_backbone_test.py::ResNetBackboneTest::test_valid_call
0.54s call     keras_cv/models/classification/image_classifier_test.py::ImageClassifierTest::test_pooling_arg_call_max_pooling
0.52s call     keras_cv/models/classification/image_classifier_test.py::ImageClassifierTest::test_valid_call
0.50s call     keras_cv/models/classification/image_classifier_test.py::ImageClassifierTest::test_pooling_arg_call_avg_pooling
0.33s call     keras_cv/models/backbones/resnet_v1/resnet_v1_backbone_test.py::ResNetBackboneTest::test_create_backbone_model_with_level_config
0.23s call     keras_cv/models/classification/image_classifier_test.py::ImageClassifierTest::test_throw_invalid_pooling

(178 durations < 0.005s hidden.  Use -vv to show these durations.)
======================================================================= short test summary info ========================================================================
FAILED keras_cv/models/classification/image_classifier_test.py::ImageClassifierTest::test_saved_model - TypeError: Function.__init__() got an unexpected keyword argument 'trainable'
========================================================= 1 failed, 77 passed, 3 skipped in 515.04s (0:08:35) ==========================================================
```

</details>

Looks like `keras_core` doesn't support `trainable` argument in `Model`'s `__init__` function. Should we remove it from our base class's `get_config` function (i.e. `Task.get_config`)?

--- a/keras_cv/models/classification/image_classifier_test.py
+++ b/keras_cv/models/classification/image_classifier_test.py
@@ -20,8 +20,8 @@ import numpy as np
 import pytest
 import tensorflow as tf
 from absl.testing import parameterized
-from tensorflow import keras
 
+from keras_cv.backend import keras
 from keras_cv.models.backbones.resnet_v2.resnet_v2_backbone import (
     ResNet18V2Backbone,
 )

